### DISCUSSION TITLE
Replace ZFS Disk - and use whole disk

### DISCUSSION THREAD

**Discussion by arogarth** [openzfs/zfs#14165] [Q&A]
We hava a zpool, created on FreeBSD years ago - FreeBSD used the whole disks - without partitioning the Disk.

If I want to replace a disk - OpenZFS/Ubuntu creates partitions on it during replacement - now the data disk is smaller than the original and I cannot detach the failed one because the sectors of the partition are less than the whole original

Is there a way to disable these partitioning on OpenZFS / Ubuntu?

Thanks!

**Comment by ghost** [comment]
No, unfortunately there is not.

**Comment by arogarth** [comment]
This is bad... :( Then there is no complete compatibility back to BSD based ZFS...

And no - I cannot destroy the pool of 200T... :/

**Comment by arogarth** [comment]
I created a Feature Request to it for better compatibility with old ZFS Pools - #14174

**Comment by ghost** [comment]
As a workaround for now you could boot into FreeBSD to do the replacement.

**Comment by pepa65** [comment]
Not sure if this helps your use case, but a workaround to getting OpenZFS to use the whole disk is:

```
cp -l /dev/sdx /dev/sdx1
zpool create pool /dev/sdx1
zpool export pool
rm /dev/sdx1
zpool import pool
```
